cli/provider/cmd.go

```diff
@@ -486,3 +482,88 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	}
 	return nil
 }
+
+func PreProcessCoverage(logger *zap.Logger, conf *config.Config) {
+	language, executable := utils.DetectLanguage(logger, conf.Command)
+	// if language is not provided through flag/config and language detected is not unknown
+	// then set the language to detected language
+	if conf.Test.Language == "" {
+		if language == models.Unknown {
+			logger.Warn("failed to detect language. please use --language to manually set the language")
+			return
+		}
+		logger.Warn(fmt.Sprintf("%s language detected. please use --language to manually set the language if needed", language))
+		conf.Test.Language = language
+	} else if language != conf.Test.Language {
+		utils.LogError(logger, nil, "language detected is different from the language provided")
+		conf.Test.SkipCoverage = true
+		return
+	}
+	var err error
+	switch conf.Test.Language {
+	case models.Python:
+		err = utils.RunCommand("coverage")
+		if err != nil {
+			conf.Test.SkipCoverage = true
+			logger.Warn("coverage tool not found. please install coverage tool using 'pip install coverage'")
```

Comment:
here to add skip coverage log and assign true to skip coverage.

Reply:
done